cpp/src/binaryop/jit/code/traits.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Copyright 2018-2019 BlazingDB, Inc.
+ *     Copyright 2018 Christian Noboa Mardini <christian@blazingdb.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace cudf {
+namespace experimental {
+namespace binops {
+namespace jit {
+namespace code {
+
+const char* traits =
+R"***(
+#pragma once
+    #include <cstdint>
+    #include <type_traits>
+
+    // -------------------------------------------------------------------------
+    // Simplifying std::is_integral
+    template <typename T>
+    constexpr bool is_integral_v = std::is_integral<T>::value;
+
+    // Simplifying std::is_floating_point
+    template <typename T>
+    constexpr bool is_floating_point_v = std::is_floating_point<T>::value;
+
+    // -------------------------------------------------------------------------
+    // type_traits cannot tell the difference between float and double
+    template <typename Type>
+    constexpr bool isFloat = false;
```

Comment:
This creates a distinction that it's a custom trait and not a standard c++ trait. In case of `is_integral_v`, it exists in a future standard of c++ (c++17, and we use c++14) but `is_float` doesn't.